packages/whenable/src/watch.js

```diff
@@ -0,0 +1,239 @@
+// @ts-check
+import { M } from '@endo/patterns';
+
+import { getWhenablePayload, unwrapPromise, basicE } from './whenable-utils.js';
+
+const { Fail } = assert;
+
+const { apply } = Reflect;
+
+export const PromiseWatcherI = M.interface('PromiseWatcher', {
+  onFulfilled: M.call(M.any()).rest(M.any()).returns(),
+  onRejected: M.call(M.any()).rest(M.any()).returns(),
+});
+
+/**
+ * @typedef {object} PromiseWatcher
+ * @property {(...args: unknown[]) => void} [onFulfilled]
+ * @property {(...args: unknown[]) => void} [onRejected]
```

Comment:
I'd prefer that the type more closely parallel the interface and the code, in that both these methods have one normal parameter (`fulfillment` and `reason`) and rest-args only for the rest. Given how unusual the rest args are, for expository purposes they should be separate from the fulfillment and reason parameters.